### ISSUE TITLE
problems finding module zope

### ISSUE THREAD

**Issue by pyinstaller-tickets-migration** [pyinstaller/pyinstaller#615]
**Original date**: 2013/08/18

my app depends on ZODB, and uses ZEO, that in turns depends on zope.event and zope.interface, which are two different eggs but use the same namespace.

```
site-packages /
    zope.event.xxxx.egg/
       zope/
          __init__.py
          event/
            __init__.py
    zope.interface.xxxx.egg/
       zope/
          __init__.py
          interface/
            __init__.py
```

the zope.event.xxx.egg/zope/**init**.py file contains :

`__import__('pkg_resources').declare_namespace(__name__)`

and zope.interface.xxx.egg/zope/**init**.py contains:

```
# this is a namespace package
try:
    import pkg_resources
    pkg_resources.declare_namespace(__name__)
except ImportError:
    import pkgutil
    __path__ = pkgutil.extend_path(__path__, __name__)
```

i tried to write a hook, but it can't pick "hook-zope.event.py" 
i tried using hiddenimports and even tried to set the pathex 
obviously the app works correctly when using from source. but when i try to package it (with pyinstaller) it says it can't find zope.event. 
"matysek_" from irc recommended me to try modulegraph and file a report.
here's what i've done :

```
>>> from modulegraph.find_modules import find_modules as fm
>>> mg = fm(['banta/__init__.py'])
>>> f = open('o.txt', 'w')
>>> for i in mg.flatten(): f.write(str(i))
...
>>> f.close()
```

and it finds it

**Comment by ypereto** [comment]
Another workaround for the zope.interface import problem is to use the following hook-zope.py file.
This is from http://svn.simo-project.org/simo/trunk/dev/pyinstaller/hooks/hook-zope.py.

```python
import os

def hook(mod):
    pth = str(mod.__path__[0])
    if os.path.isdir(pth):
        owd = os.getcwd()
        print 'started from', owd
        print 'manipulating', pth
        os.chdir(pth)
        os.chdir('../..')
        cwd = os.getcwd()
        print 'moved to', cwd
        zopedirs = [''.join((cwd, os.sep, item, os.sep, 'zope')) \
                    for item in os.listdir('.') \
                    if 'zope' in item and item != 'zope' \
                       and os.path.isdir(item)]
        if os.path.exists('zope'):
            zopedirs.append(os.path.join(cwd, 'zope'))
        mod.__path__ = zopedirs
        print 'zope.__path__ fixed to:'
        for path in mod.__path__:
            print path
        print 'returning to', owd
        os.chdir(owd)
    return mod
```

**Comment by htgoebel** [comment]
@ypereto Why do you want to use this work-around, as this is already solved (see https://github.com/pyinstaller/pyinstaller/issues/615#issuecomment-59615701) Does this not work for you?

**Comment by ypereto** [comment]
I'm using pyinstaller v2.1, which does not have this fix. I'm assuming this will be fixed in 2.2. I'll be happy to dump the hook once v2.2 is released.

**Comment by matysek** [comment]
@ypereto This will be fixed in 3.0.   2.2 won't be released.
